### ISSUE TITLE
Support for IOS

### ISSUE THREAD

**Issue by AndreyPootMay** [ivadim/fruitnanny#7]
Hello, I tried to perform the same steps of the installation and it works correctly, but I see that they do not work within the _**ios safari**_ browser.

How could I fix this problem?

Thank you

**Comment by jameshope87** [comment]
Unfortunately apple is just way behind the times.

To fix the issue either swap to another device or campaign for apple to implement WebRTC.

**Comment by dangentile** [comment]
iOS should support WebRTC as of iOS 11/12 however this code does not seem to work in iOS/Safari. I wonder how much trouble it might be to update and get it working

**Comment by ivadim** [comment]
Hello,
I don't have any iOS device, so, can't test it. Under the hood Fruitnanny uses https://github.com/meetecho/janus-gateway . If they support iOS then it should work with this project as well.

**Comment by AndreyPootMay** [comment]
> Hello,
> I don't have any iOS device, so, can't test it. Under the hood Fruitnanny uses https://github.com/meetecho/janus-gateway . If they support iOS then it should work with this project as well.

You Only uses "Janus.js" API in the app code?

**Comment by dangentile** [comment]
I have updated the Janus build to the latest master, also updated the janus.js and adapter.js to latest versions. Finally I added "playsinline" to the end video tag in the html. I have been able to get the video to stream inconsistently to Safari on iOS (have to refresh the page a number of times) and no audio. I plan to mess around more with it next week when I have more time. I don't really know much about programming, or webRTC, learning as I go, but I think adding support for iOS would be pretty simple for someone more experienced than I.

**Comment by ivadim** [comment]
@AndreyPootMay yes it uses janus.js: https://github.com/meetecho/janus-gateway/blob/master/html/janus.js and forked version of https://github.com/meetecho/janus-gateway/blob/master/html/streamingtest.html

@dangentile thank you. Looks like it's a good progress. Feel free to make a PR to contribute your work back.

**Comment by Gudui** [comment]
@dangentile did you make any further progress with the IOS web streaming?

**Comment by dangentile** [comment]
@Gudui 

no I was not able to make work with iOS and Janus after more than a few attempts, but I am not a programmer/web dev... I did however find a similar project as was able to get an HLS audio video stream working to iOS devices. Check out this link

https://github.com/kamranayub/kamranayub.github.io/blob/source/input/guides/raspberry-pi-3-baby-monitor.md

I added some additional features to this build, second mic to do alerts when baby is crying, and I went with the blink stick nano for the light for now, also setup the Pi to create a hotspot when no wifi is found to be used when traveling. I plan to put a full guide together later (once things settle down now that baby is here, and I have used the monitor to work out any kinks)

**Comment by ivadim** [comment]
Hi Gudui, did you setup fruitnanny recently?
I added some info month ago here: https://ivadim.github.io/2017-08-21-fruitnanny/ 
Search for "Possible Fix for Safary". I managed to make video working on iOS, audio still has problem though :(

**Comment by progware** [comment]
Hi guys,
I have successfully setup the project and it works fine. I have changed few things and added some extra sensor. I also applied iOS/OSX fix. And here is my observation:

Before using fix for Safari:
- at the beginning it worked just fine with OSX, but not using Safari
- it was not working on iPhone 5 and 5S (just temperature and humidity)
- worked on all Windows and Android devices I have

After using fix for Safari:
- still works with Android devices, Windows and OSX Mozilla Firefox
- started to work on Safari, stopped working with Chrome in OSX with 'No WebRTC support...' message, but after some time (probably some autoupdate was done by Chrome), I read that some settings for webRTC are in chrome://flags, tried to use it but no success.
- started to work with iOS devices that I have  (5 and 5E), but with exception that video and audio is loaded once and next time I reload page it does not work*, actually my wife uses one of this iPhones and she says that 'new solution does not work', before I implemented it using Flask. 

*To make it work I had to go and enable some experimental settings in iPhone Settings > Safari > Advanced > Experimental Features (this my be different as I do not use iPhone and cannot check when I write this post)(I am not sure which ones I enabled, those with WebRTC and some other possible blockers)


I did not have much time yet to debug it on OSX Chrome and iPhone, but as soon as I get there I add some info to this issue.

My question is:
Did anyone have similar problems with Chrome on OSX or iPhone devices? It works from time to time. Once it worked for me when I switched to anonymous mode in iPhone but then it stopped working. Maybe there is some problem with Safari version ?

Best Regards
Piotr
